Editing/input-event-dataTransfer-explainer.md

```diff
@@ -0,0 +1,158 @@
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+## Table of Contents
+
+- [Introduction](#introduction)
+- [User-Facing Problem](#user-facing-problem)
+- [Goals](#goals)
+- [Non-goals](#non-goals)
+- [Motivation](#motivation)
+  - [Code Example](#code-example)
+    - [Before the Fix](#before-the-fix)
+    - [After the Fix](#after-the-fix)
+- [Considered Alternatives](#considered-alternatives)
+- [Security and Privacy](#security-and-privacy)
+- [Performance Impact](#performance-impact)
+- [Interoperability](#interoperability)
+- [References and Acknowledgements](#references-and-acknowledgements)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+
+## Introduction
+
+Modern web applications rely heavily on rich text editing experiences, especially within [`contenteditable`](https://html.spec.whatwg.org/multipage/interaction.html#attr-contenteditable) elements. [`InputEvent.dataTransfer`](https://w3c.github.io/input-events/#dom-inputevent-datatransfer) is a property that provides access to a [`dataTransfer`](https://html.spec.whatwg.org/multipage/dnd.html#datatransfer) object during input events triggered by paste, drop, or replacement actions. It allows developers to inspect and handle the data being inserted. However, developers have long faced limitations when handling paste and drop operations due to the absence of the [`dataTransfer`](https://html.spec.whatwg.org/multipage/dnd.html#datatransfer) property on [Input Event Types](https://w3c.github.io/input-events/#overview) objects. This explainer proposes that [`InputEvent.dataTransfer`](https://w3c.github.io/input-events/#dom-inputevent-datatransfer) be exposed for specific input types [`insertFromPaste`, `insertReplacementText`, and `insertFromDrop`](https://w3c.github.io/input-events/#overview) within [`contenteditable`](https://html.spec.whatwg.org/multipage/interaction.html#attr-contenteditable) contexts. This change puts Chromium in alignment with the [`W3C spec`](https://www.w3.org/TR/input-events-2/).
+
+## User-Facing Problem
+A user pastes formatted content (e.g., bold text, lists, links) into a custom editor. The developers listening to the input event with inputType = "insertFromPaste" in the following example couldn’t access [`InputEvent.dataTransfer`](https://w3c.github.io/input-events/#dom-inputevent-datatransfer) — it was null.
+
+```html
+<p contenteditable="true">
+  Go on, try pasting some content into this editable paragraph and see what
+  happens!
+</p>
+
+<p class="result"></p>
+```
+
+```js
+const editable = document.querySelector("p[contenteditable]");
+const result = document.querySelector(".result");
+editor.addEventListener("input", (event) => {
+  result.textContent = event.dataTransfer.getData("text/html"); // Fails as error - Cannot read properties of null (reading 'getData')
```

Comment:
This example is kind of odd because here `event.dataTransfer` is used without a check, and then below it's null-checked. Maybe this line should be merged with the line below in the `if` check that also gets `event.dataTransfer.getData("text/html")`?